counter/src/main/java/org/infinispan/tutorial/simple/counter/InfinispanCounter.java

```diff
@@ -0,0 +1,97 @@
+package org.infinispan.tutorial.simple.counter;
+
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.counter.EmbeddedCounterManagerFactory;
+import org.infinispan.counter.api.CounterManager;
+import org.infinispan.counter.api.StrongCounter;
+import org.infinispan.counter.api.WeakCounter;
+import org.infinispan.counter.configuration.CounterManagerConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+
+/**
+ * The Counter simple tutorial.
+ * <p>
+ * The counters are available in Infinispan since version 9.1
+ *
+ * @author Pedro Ruivo
+ */
+public class InfinispanCounter {
+
+   public static void main(String[] args) throws Exception {
+      // Setup up a clustered cache manager
+      GlobalConfigurationBuilder global = GlobalConfigurationBuilder.defaultClusteredBuilder();
+      // Create the counter configuration builder
+      CounterManagerConfigurationBuilder builder = global.addModule(CounterManagerConfigurationBuilder.class);
+
+      // Create 3 counters.
+      // The first counter is bounded to 10 (upper-bound).
+      builder.addStrongCounter().name("counter-1").upperBound(10).initialValue(1);
+      // The second counter is unbounded
+      builder.addStrongCounter().name("counter-2").initialValue(2);
+      // And finally, the third counter is a weak counter.
+      builder.addWeakCounter().name("counter-3").initialValue(3);
+      // Initialize the cache manager
+      DefaultCacheManager cacheManager = new DefaultCacheManager(global.build());
+
+      // Retrieve the CounterManager from the CacheManager. Each CacheManager has it own CounterManager
+      CounterManager counterManager = EmbeddedCounterManagerFactory.asCounterManager(cacheManager);
+
+      // StrongCounter provides the higher consistency. Its value is known during the increment/decrement and it may be bounded.
+      // Bounded counters are aimed for uses cases where a limit is needed.
+      StrongCounter counter1 = counterManager.getStrongCounter("counter-1");
+      // All methods returns a CompletableFuture. So you can do other work while the counter value is being computed.
+      counter1.getValue().thenAccept(value -> System.out.println("Counter-1 initial value is " + value)).get();
+
+      // Try to add more than the upper-bound
+      counter1.addAndGet(10).handle((value, throwable) -> {
+         // Value is null since the counter is bounded and we can add 10 to it.
+         System.out.println("Counter-1 Exception is " + throwable.getMessage());
+         return 0;
+      }).get();
+
+      // Check the counter value. It should be the upper-bound (10)
```

Comment:
This behaviour is not specified in the `getValue` javadoc. The method's javadoc should mention that if an attempt has been to go pass the upper bound, next time `getValue` is called, the upper bound would be returned.

The reason I think this is important is because one could expect that if an exception is thrown after trying to update a count beyond its upper bound, the value of the counter would be the value before trying to overflow it?